### PR TITLE
feat: add the third step of sign up relates #118

### PR DESCRIPTION
**Pull request by naghamabuwarda**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,5 +29,8 @@
     "@commitlint/cli": "^17.1.2",
     "@commitlint/config-conventional": "^17.1.0",
     "husky": "^8.0.1"
+  },
+  "dependencies": {
+    "@mui/lab": "^5.0.0-alpha.105"
```
Comment (LinaYahya) [inline package.json:34]: It's already installed at `client`

File: server/src/controllers/jobs.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ const addJob = async (req: Request, res: Response) => {
   return { status: 201, data: job };
 };
 const searchJobs = async (req: Request) => {
-  const JOBS_PER_PAGE = 8;
+  const JOBS_PER_PAGE = 5;
```
Comment (LinaYahya) [inline server/src/controllers/jobs.ts:53]: Not sure why is that?

File: client/src/validation/index.ts [hunk 3]
```diff
@@ -46,7 +46,29 @@ const signUpSchema = yup.object({
     .oneOf([yup.ref('password')], 'The field must match')
     .required('The confirm password field is required'),
 });
+const thirdStepValidation = yup.object({
+  Title: yup
```
Comment (LinaYahya) [inline client/src/validation/index.ts:50]:
```suggestion
  title: yup // as it is in the freelancer model
```

File: client/src/validation/index.ts [hunk 4]
```diff
@@ -46,7 +46,29 @@ const signUpSchema = yup.object({
...
+    .string()
+    .required('Title is required'),
+  Major: yup
```
Comment (LinaYahya) [inline client/src/validation/index.ts:53]:
```suggestion
major: yup
```

File: client/src/validation/index.ts [hunk 5]
```diff
@@ -46,7 +46,29 @@ const signUpSchema = yup.object({
...
+    .string()
+    .required('Major is required'),
+  portfolio: yup
+    .string()
+    .optional()
+    .nullable()
+    .url(),
+  description: yup
```
Comment (LinaYahya) [inline client/src/validation/index.ts:61]:
```suggestion
  brief: yup
```

File: client/src/pages/Signup/index.tsx [hunk 6]
```diff
@@ -3,7 +3,7 @@ import {
   Box, Step, Stepper, StepLabel, Button,
 } from '@mui/material';
 import './style.css';
-import { FirstStep, Signup } from '../../components';
+import { FirstStep, Signup, ThirdStepSignUp } from '../../components';
```
Comment (LinaYahya) [inline client/src/pages/Signup/index.tsx:6]:
```suggestion
import { FirstStep, Signup, ThirdStepSignUp } from '../../components';
```
I think we can give it more descriptive name

File: client/src/components/signup/index.tsx [hunk 7]
```diff
@@ -100,4 +106,127 @@ function Signup({ setActiveStep, userRole }: SignupProps) {
     </div>
   );
 }
-export default Signup;
+function ThirdStepSignUp() {
```
Comment (LinaYahya) [inline client/src/components/signup/index.tsx:109]: Why not to separate in another file!

File: client/src/components/signup/index.tsx [hunk 8]
```diff
@@ -100,4 +106,127 @@ function Signup({ setActiveStep, userRole }: SignupProps) {
...
+              >
+                <CloudUploadIcon />
+                Upload image
+                <input hidden accept="image/*" multiple type="file" />
+              </LoadingButton>
+              <FormHelperText id="error-text" error>
+                {formik.touched.image && formik.errors.image}
+              </FormHelperText>
+            </label>
+            <div style={{
```
Comment (LinaYahya) [inline client/src/components/signup/index.tsx:213]: why not to give it a classname and then style it in the `css` file

**Review by LinaYahya — CHANGES_REQUESTED** [review]
Good Job @naghamabuwarda ,
I've noticed that we create image as required in the `Freelancer` model,
Can you fix it please?

**Review by LinaYahya — APPROVED** [review]

**Review by mohammedagha27 — APPROVED** [review]

**Review by shathakh — APPROVED** [review]

**Review by alishanti98 — APPROVED** [review]

**Pull request merged**